### PR TITLE
Use lewton

### LINKED ISSUE #74: Switch to lewton
https://github.com/est31/lewton

### PR DESCRIPTION
**Pull request by est31**
Fixes #74.

### CODE REVIEW

File: src/decoder/vorbis.rs [hunk 1]
```diff
@@ -4,34 +4,42 @@ use std::vec;
...
 
-        let (data, rate, channels) = match decoder.packets().filter_map(Result::ok).next() {
-            Some(p) => (p.data, p.rate as u32, p.channels as u16),
-            None => (Vec::new(), 44100, 2),
+        let mut data = match stream_reader.read_dec_packet_itl().ok().and_then(|v| v) {
+            Some(d) => d,
+            None => Vec::new(),
+        };
+
+        match stream_reader.read_dec_packet_itl().ok().and_then(|v| v) {
```
Comment (tomaka) [inline src/decoder/vorbis.rs]: Why call `read_dec_packet_itl()` three times? Is it an arbitrary choice?

File: src/decoder/vorbis.rs [hunk 2]
```diff
@@ -44,12 +52,12 @@ impl<R> Source for VorbisDecoder<R> where R: Read + Seek {
 
     #[inline]
     fn get_channels(&self) -> u16 {
-        self.current_channels
+        self.stream_reader.ident_hdr.audio_channels as u16
```
Comment (tomaka) [inline src/decoder/vorbis.rs]: I thought that in the vorbis format, it was possible that the number of channels and samples rate changes between two frames.

File: src/decoder/vorbis.rs (hunk 1 above)
Comment (est31) [inline src/decoder/vorbis.rs]: Ah sorry, that was left from when I was debugging. Will remove

File: src/decoder/vorbis.rs (hunk 2 above)
Comment (est31) [inline src/decoder/vorbis.rs]: No, the number of channels is constant over the whole stream: https://www.xiph.org/vorbis/doc/Vorbis_I_spec.html#x1-120001.2.1

**Comment by tomaka** [comment]
Thanks!

**Pull request merged**